mods/ra/maps/allies-05a/allies05a-AI.lua

```diff
@@ -243,25 +243,32 @@ ProduceAircraft = function()
 	end
 
 	ussr.Build(SovietAircraftType, function(units)
-		Yaks[#Yaks + 1] = units[1]
+		local yak = units[1]
+		Yaks[#Yaks + 1] = yak
 
 		if #Yaks == 2 then
-			Trigger.OnKilled(units[1], ProduceAircraft)
+			Trigger.OnKilled(yak, ProduceAircraft)
 		else
 			Trigger.AfterDelay(DateTime.Minutes(1), ProduceAircraft)
 		end
 
 		local target = nil
-		Trigger.OnIdle(units[1], function()
+		Trigger.OnIdle(yak, function()
 			if not target or target.IsDead or (not target.IsInWorld) then
 
 				local enemies = Utils.Where(Map.ActorsInWorld, function(self) return self.Owner == greece and self.HasProperty("Health") end)
 				if #enemies > 0 then
 					target = Utils.Random(enemies)
-					units[1].Attack(target)
+				else
+					yak.Wait(DateTime.Seconds(5))
 				end
+			end
+
+			if yak.HasAmmo() then
```

Comment:
I guess this should be `if target and yak.HasAmmo() then`, just to be safe. `target` is guaranteed to be set here (although it probably will be).